python/cudf/cudf/core/index.py

```diff
@@ -334,13 +334,16 @@ def __contains__(self, item):
             item, tuple(np.sctypes["int"] + np.sctypes["float"] + [int, float])
         ):
             return False
+        elif isinstance(item, bool):
+            return False
         try:
-            item = pd.core.dtypes.common.ensure_python_int(item)
-        except TypeError:
+            int_item = int(item)
+            assert int_item == item
+        except (TypeError, AssertionError):
             return False
-        if not item % 1 == 0:
+        if not int_item % 1 == 0:
             return False
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏
```suggestion
```

This is an unreachable branch since $\\{x: x \in \mathbb{Z} \land x \mod 1 \ne 0\\} = \emptyset$